src/pages/MenuPage.js

```diff
@@ -38,107 +86,109 @@ export class MenuPage extends React.Component {
-              text={navStrings.customer_invoices}
-              onPress={() => navigateTo('customerInvoices', navStrings.customer_invoices)}
-            />
-            <Button
-              style={globalStyles.menuButton}
-              textStyle={globalStyles.menuButtonText}
-              text={navStrings.customer_requisitions}
-              onPress={() => navigateTo('customerRequisitions', navStrings.customer_requisitions)}
-            />
-          </View>
-
-          <View style={[localStyles.container, localStyles.centralContainer]}>
-            <Image
-              style={localStyles.image}
-              resizeMode="contain"
-              // eslint-disable-next-line global-require
-              source={require('../images/menu_truck.png')}
-            />
-            <Button
-              style={globalStyles.menuButton}
-              textStyle={globalStyles.menuButtonText}
-              text={navStrings.supplier_invoices}
-              onPress={() => navigateTo('supplierInvoices', navStrings.supplier_invoices)}
-            />
-            <Button
-              style={globalStyles.menuButton}
-              textStyle={globalStyles.menuButtonText}
-              text={navStrings.supplier_requisitions}
-              onPress={() => navigateTo('supplierRequisitions', navStrings.supplier_requisitions)}
-            />
-            {isInAdminMode && (
-              <Button
-                style={globalStyles.menuButton}
-                textStyle={globalStyles.menuButtonText}
-                text="Export Data"
-                onPress={this.exportData}
-              />
-            )}
-          </View>
-
-          <View style={localStyles.container}>
-            <Image
-              style={localStyles.image}
-              resizeMode="contain"
-              // eslint-disable-next-line global-require
-              source={require('../images/menu_pc_clipboard.png')}
-            />
-            <Button
-              style={globalStyles.menuButton}
-              textStyle={globalStyles.menuButtonText}
-              text={navStrings.current_stock}
-              onPress={() => navigateTo('stock', navStrings.current_stock)}
-            />
-            <Button
-              style={globalStyles.menuButton}
-              textStyle={globalStyles.menuButtonText}
-              text={navStrings.stocktakes}
-              onPress={() => navigateTo('stocktakes', navStrings.stocktakes)}
-            />
-            {isInAdminMode && (
-              <Button
-                style={globalStyles.menuButton}
-                textStyle={globalStyles.menuButtonText}
-                text="Realm Explorer"
-                onPress={() => navigateTo('realmExplorer', 'Database Contents')}
-              />
-            )}
-          </View>
-        </View>
-        <View style={globalStyles.bottomContainer}>
+      <View style={pageContentContainer}>
+        {this.layout.map(row => (
```

Comment:
need a `key` property on all of these `.map()` rendering I think? (how do you feel about using `generateUUID` for `key` props?)

Reply:
good point, fixed: https://github.com/openmsupply/mobile/pull/868/commits/166d3695533183bb3e28133c994b054d7510490e